api/http/security/authorization.go

```diff
@@ -160,3 +163,44 @@ func AuthorizedAccess(userID portainer.UserID, memberships []portainer.TeamMembe
 
 	return false
 }
+
+func UpdateUserServiceAccountsForEndpoint(endpointID portainer.EndpointID, dataStore dataservices.DataStore, k8sClientFactory *cli.ClientFactory) {
```

Comment:
Consider using a receiver. i.e. can then be called like `endpoint.UpdateUserServiceAccounts(dataStore, k8sClientFactory) `

```
func (endpointID portainer.EndpointID)UpdateServiceAccounts(dataStore ...
```